go.mod

```diff
@@ -26,9 +27,9 @@ require (
 	github.com/stretchr/testify v1.10.0
 	go.uber.org/zap v1.27.0
 	golang.org/x/sync v0.12.0
-	k8s.io/api v0.32.3
-	k8s.io/apimachinery v0.32.3
-	k8s.io/client-go v0.32.3
+	k8s.io/api v0.33.0
```

Comment:
Can we bump the Kubernetes libraries in a different PR? This requires more changes, like updating the KIND versions for the tests, among other things